pkg/webhooks/controlplanemachineset/webhooks.go

```diff
@@ -255,7 +255,11 @@ func validateOpenShiftMachineV1BetaTemplate(parentPath *field.Path, template mac
 func validateOpenShiftMachineV1BetaTemplateOnCreate(parentPath *field.Path, template machinev1.OpenShiftMachineV1Beta1MachineTemplate, machines []machinev1beta1.Machine) []error {
 	errs := []error{}
 
-	errs = append(errs, checkOpenShiftFailureDomainsMatchMachines(parentPath.Child("failureDomains"), template.FailureDomains, machines)...)
+	if template.FailureDomains.Platform == "" {
+		errs = append(errs, checkOpenShiftProviderSpecFailureDomainMatchesMachines(parentPath.Child("template.providerSpec"), template, machines)...)
```

Comment:
This should be a list of strings
```suggestion
		errs = append(errs, checkOpenShiftProviderSpecFailureDomainMatchesMachines(parentPath.Child("template", "providerSpec"), template, machines)...)
```